uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/CamelEditorTest.java

```diff
@@ -263,4 +265,36 @@ public void testDragAndDropComponents() {
 		assertTrue(EditorManipulator.isEditorContentEqualsFile("resources/camel-context-all.xml"));
 		assertTrue(LogGrapper.getPluginErrors("fuse").size() == 0);
 	}
+	
+	/**
+	 * <p>
+	 * Test checks collapse/expand feature of Camel editor components
+	 * </p>
+	 * <ol>
+	 * <li>create a new project from template 'Content Based Router'</li>
+	 * <li>open camel-context.xml file</li>
+	 * <li>Invoke 'Collapse' on route component 'Choice'</li>
+	 * <li>Check if component was collapsed (check if changed size of collapsed component area and height is smaller
+	 * then begin values)</li>
+	 * <li>Check fuse errors in Error Log View</li>
+	 * <li>Invoke 'Expand' on route component 'Choice'</li>
+	 * <li>Check if component was expanded (check changed size of expanded component area -> should be same as area size
+	 * and height before collapse)</li>
+	 * <li>Check fuse errors in Error Log View</li>
+	 * </ol>
+	 */
+	@Test
+	public void testCollapseExpandFeature() {
+		CamelEditor editor = new CamelEditor("camel-context.xml");
+		CamelComponentEditPart editPart = new CamelComponentEditPart("Choice");
+		int startHeight = editPart.getBounds().height;
+		editor.doOperation("Choice", "Collapse");
+		AbstractWait.sleep(TimePeriod.MEDIUM);
```

Comment:
I think that it is best to use WaitUntil, it will help to have faster test execution.
i think it will be something like:
new WaitUntil(new AbstractWaitCondition() {
			
			@Override
			public boolean test() {
				return editPart.getBounds().height < startHeight;
			}
			
			@Override
			public String description() {
				return "Route component 'Choice' wasn't collapsed properly";
			}
			
		}, TimePeriod.MEDIUM);